lib/v2/screens/sign_up/claim_invite/claim_invite_screen.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:seeds/utils/string_extension.dart';
+import 'package:seeds/v2/components/scanner/scanner_widget.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/screens/sign_up/claim_invite/components/bottom_container.dart';
+import 'package:seeds/v2/screens/sign_up/viewmodels/bloc.dart';
+import 'package:seeds/v2/screens/sign_up/viewmodels/states/claim_invite_state.dart';
+
+class ClaimInviteScreen extends StatefulWidget {
+  @override
+  _ClaimInviteScreenState createState() => _ClaimInviteScreenState();
+}
+
+class _ClaimInviteScreenState extends State<ClaimInviteScreen> {
+  late SignupBloc _signupBloc;
+  late ClaimInviteState _currentState;
+
+  late ScannerWidget _scannerWidget;
+
+  @override
+  void initState() {
+    super.initState();
+    _scannerWidget = ScannerWidget(resultCallBack: _onQRScanned);
+    _signupBloc = BlocProvider.of<SignupBloc>(context);
+    _currentState = _signupBloc.state.claimInviteState;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(
+          'Scan QR Code',
+          style: Theme.of(context).textTheme.subtitle1,
+        ),
+      ),
+      body: BlocListener<SignupBloc, SignupState>(
+        listenWhen: (previousState, currentState) => previousState != currentState,
+        listener: (context, state) {
+          _currentState = state.claimInviteState;
+
+          if (_currentState.pageState == PageState.success && !_currentState.inviteMnemonic.isNullOrEmpty) {
```

Comment:
i would have modeled this a little different, more purpose on the commands 

in the state have a PageCommand 

 listenWhen: (previousState, currentState) => previousState != currentState,
 would be  listenWhen: (previousState, currentState) => pageCommand != null,
 
 class StartScan extends  PageCommand {}
 class StopScan extends  PageCommand {}
 
 as my commands